docs/userguide/networking/work-with-networks.md

```diff
@@ -0,0 +1,454 @@
+172.17.0.3	498eaaaf328e
+127.0.0.1	localhost
+::1	localhost ip6-localhost ip6-loopback
+fe00::0	ip6-localnet
+ff00::0	ip6-mcastprefix
+ff02::1	ip6-allnodes
+ff02::2	ip6-allrouters
+172.21.0.3	container3
+172.21.0.3	container3.isolated_nw
+```
+
+On the `isolated_nw` which was user defined, the Docker network feature updated the `/etc/hosts` with the proper name resolution.  Inside of `container2` it is possible to ping `container3` by name.
+
+```bash
+/ # ping -w 4 container3
+PING container3 (172.21.0.3): 56 data bytes
+64 bytes from 172.21.0.3: seq=0 ttl=64 time=0.070 ms
+64 bytes from 172.21.0.3: seq=1 ttl=64 time=0.080 ms
+64 bytes from 172.21.0.3: seq=2 ttl=64 time=0.080 ms
+64 bytes from 172.21.0.3: seq=3 ttl=64 time=0.097 ms
+
+--- container3 ping statistics ---
+4 packets transmitted, 4 packets received, 0% packet loss
+round-trip min/avg/max = 0.070/0.081/0.097 ms
+```
+
+This isn't the case for the default bridge network. Both `container2` and  `container1` are connected to the default bridge network. Docker does not support automatic service discovery on this network. For this reason, pinging  `container1` by name fails as you would expect based on the `/etc/hosts` file:
+
+```bash
+/ # ping -w 4 container1
+ping: bad address 'container1'
+```
+
+A ping using the `container1` IP address does succeed though:
+
+```bash
+/ # ping -w 4 172.17.0.2
+PING 172.17.0.2 (172.17.0.2): 56 data bytes
+64 bytes from 172.17.0.2: seq=0 ttl=64 time=0.095 ms
+64 bytes from 172.17.0.2: seq=1 ttl=64 time=0.075 ms
+64 bytes from 172.17.0.2: seq=2 ttl=64 time=0.072 ms
+64 bytes from 172.17.0.2: seq=3 ttl=64 time=0.101 ms
+
+--- 172.17.0.2 ping statistics ---
+4 packets transmitted, 4 packets received, 0% packet loss
+round-trip min/avg/max = 0.072/0.085/0.101 ms
+```
+
+If you wanted you could link `container1` to `container2` and that would enable the two containers to interact by name as well as IP.
+
+Detach from a `container2` and leave it running using `CTRL-p CTRL-q`.
+
+In this example, `container2` is attached to both networks and so can talk to
+`container1` and `container3`. But `container3` and `container1` are not in the
+same network and cannot communicate. Test, this now by attaching to
+`container3` and attempting to ping `container1` by IP address.
+
+```bash
+$ docker attach container3
+/ # ping 172.17.0.2
+PING 172.17.0.2 (172.17.0.2): 56 data bytes
+^C
+--- 172.17.0.2 ping statistics ---
+10 packets transmitted, 0 packets received, 100% packet loss
+
+```
+
+To connect a container it must be running. If you stop a container and inspect a network it belongs to, you won't see that container.  The `docker network inspect` command only shows running containers.
+
+## Disconnecting containers
+
+You can disconnect a container from a network using the `docker network disconnect` command.
+
+```
+$ docker network disconnect isolated_nw container2
+
+docker inspect --format='{{json .NetworkSettings.Networks}}'  container2 | python -m json.tool
```

Comment:
Hm, also wonder if `--format` could have a built-in option to pretty-print JSON, that'd be cool (I'll ask / create an issue)